config/v1/types_ingress.go

```diff
@@ -172,6 +180,98 @@ type ComponentRouteStatus struct {
 	RelatedObjects []ObjectReference `json:"relatedObjects"`
 }
 
+// RequiredRouteAnnotations specifies annotations that are required to be set on
+// newly created routes matching some criteria.
+type RequiredRouteAnnotations struct {
+	// domains is an optional list of domains for which these annotations
+	// are required.  If domains is specified and a route is created with a
+	// spec.host matching one of the domains, the route must specify the
+	// annotations specified in requiredAnnotations.  If domains is empty,
+	// the specified annotations are required for all newly created routes.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Domains string `json:"domains,omitempty"`
+
+	// policy specifies the policy for user-provided annotation values on
+	// newly created routes.  The following values are allowed for this
+	// field:
+	//
+	// * "Allow" allows the user to provide arbitrary annotation values.
+	//   With this policy, if an annotation key in requiredAnnotations is
+	//   not specified on the route at all, the route is rejected.  If every
+	//   annotation key in requiredAnnotations is specified, then the route
+	//   is admitted, irrespective of the annotation values.
+	//
+	// * "AllowWithDefaulting" allows the user to provide arbitrary
+	//   annotation values, and sets defaults if an annotation is missing.
+	//   With this policy, if an annotation key in requiredAnnotations is
+	//   not specified on the route at all, the annotation from
+	//   requiredAnnotations (key and value) is added.
+	//
+	// * "Deny" prohibits the user from specifying annotation values that
+	//   differ from those in requiredAnnotations.  With this policy, if an
+	//   annotation in requiredAnnotations is not specified with the same
+	//   key *and* value on the route, the route is rejected.
+	//
+	// * "Override" overrides user-provided annotations with the annotations
+	//   in requiredAnnotations.  With this policy, if an annotation in
+	//   requiredAnnotations is not specified on the route, the annotation
+	//   is copied from requiredAnnotations is used, and if an annotation
+	//   key in requiredAnnotations is specified on the route, the
+	//   annotation's value is overridden with the value from
+	//   requiredAnnotations.
+	//
+	// Note that the "AllowWithDefaulting" and "Override" options are
+	// dangerous to use.  With these options, a route that is created using
+	// a particular definition one cluster may behave differently from a
+	// route that is created using the same definition but on a cluster with
+	// different required route annotations configured.  Using "Allow" or
+	// "Deny" is safer because either option causes route creation to fail,
+	// loudly, if an annotation is not set as expected for the cluster.
+	Policy RequiredRouteAnnotationsPolicy `json:"policy"`
```

Comment:
I think that if you narrow to just hsts as mentioned in comment https://github.com/openshift/api/pull/881/files#r605718112, you don't need to have a generic policy specified.  I think HSTS has a few useful dimensions to control with a few logical directions to have flexibility in